### PR TITLE
58 update unit tests

### LINKED ISSUE #58: Update Unit tests
The unit tests need to be updated to reflect the changes in the data stores

### LINKED ISSUE #62: Issue where duplicate entries in DB
When the `KnownPosts` table gets synchronized duplicate entries get inserted for the posts.

### PR DESCRIPTION
**Pull request by MarshallAsch**
closes #58. Fixed unit tests, the DB test no longer depends on existing data in the database.
There were 2 db migrations to fix the duplicate records being entered into the known posts table. This also fixes #62 where the db entries got inserted multiple times.

### CODE REVIEW

File: app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java [hunk 1]
```diff
@@ -68,26 +80,34 @@ public void checkBlocked()
...
-    public void insertKnownPost(){
-        //null comment insert test should return true
+    public void insertKnownPost()
+    {
         assertTrue(database.insertKnownPost("123456789012345678901234567890123456", null));
+        // insert a duplicate should fail
+        assertFalse(database.insertKnownPost("123456789012345678901234567890123456", null));
         //empty comment string should return true
-        assertTrue(database.insertKnownPost("123456789012345678901234567890123456", ""));
+        assertTrue(database.insertKnownPost("HASH_2789012345678901234567890123456", ""));
```
Comment (weihanli101) [inline app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java]: This line and the line below. Why does the test hash start with 'HASH'?

File: app/src/main/java/ca/marshallasch/veil/database/Database.java [hunk 2]
```diff
@@ -125,6 +135,10 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         if (oldVersion < 4) {
             Migrations.upgradeV4(db);
         }
+
+        if (oldVersion < 6) {
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/database/Database.java]: What happened to v5?

File: app/src/main/java/ca/marshallasch/veil/database/KnownPostsContract.java [hunk 3]
```diff
@@ -13,10 +13,13 @@
...
-    static final String SQL_CREATE_KNOWN_POSTS = "CREATE TABLE " +  KnownPostsEntry.TABLE_NAME + "(" +
-            KnownPostsEntry._ID + " INTERGER PRIMARY KEY," +
-            KnownPostsEntry.COLUMN_COMMENT_HASH + " VARCHAR(36) UNIQUE," +
-            KnownPostsEntry.COLUMN_POST_HASH + " VARCHAR(36))";
+    static final String SQL_CREATE_KNOWN_POSTS = "CREATE TABLE " +  KnownPostsEntry.TABLE_NAME +
+            "(" +
+            KnownPostsEntry._ID + " INTEGER PRIMARY KEY," +
+            KnownPostsEntry.COLUMN_COMMENT_HASH + " VARCHAR(36)," +
+            KnownPostsEntry.COLUMN_POST_HASH + " VARCHAR(36)," +
+            "UNIQUE(" + KnownPostsEntry.COLUMN_POST_HASH + ", " + KnownPostsEntry.COLUMN_COMMENT_HASH + ")" +
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/database/KnownPostsContract.java]: Why is this line added and why was UNIQUE removed from COLUMN_COMMENT_HASH

File: app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java (hunk 1 above)
Comment (MarshallAsch) [reply app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java]: it starts with hash so that they are different strings, I try to make them all different when the need to be but also predictable so that there are not duplicates in different areas

File: app/src/main/java/ca/marshallasch/veil/database/KnownPostsContract.java (hunk 3 above)
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/database/KnownPostsContract.java]: it need to be removed because it is not the comment hash that is unique it is the combination of post and comment hashes, also if you insert null's then the UNIQUE constraint gets ignored

File: app/src/main/java/ca/marshallasch/veil/database/Database.java (hunk 2 above)
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/database/Database.java]: I messed up the migration again and had to go to v6 I meant to insert a comment but forgot about it.

**Comment by weihanli101** [comment]
Looks good
